src/app/marketplace/edit-ad.component.ts

```diff
@@ -0,0 +1,329 @@
+      promotionalEndDate: adManage.promotionalPeriod ? adManage.promotionalPeriod.end : null,
+      promotionalPrice: adManage.promotionalPrice,
+      description: [adManage.description, Validators.required],
+      addresses: adManage.addresses,
+      version: adEdit.version,
+      id: this.id
+    });
+    this.form.setControl('customValues', this.fieldHelper.customValuesFormGroup(data.customFields, {
+      currentValues: adManage.customValues
+    }));
+    this.addSub(this.form.get('currency').valueChanges.subscribe(id => this.updateCurrency(id, data)));
+    // Preselect the first currency when creating a new ad
+    // or use the single returned currency when editing
+    if (!empty(data.currencies)) {
+      this.currency = data.currencies[0];
+    }
+    this.uploadedImages = [];
+  }
+
+  resolveMenu(data: AdBasicData) {
+    return this.authHelper.userMenu(data.owner, Menu.SEARCH_ADS);
+  }
+
+  /**
+   * Changes the current currency updating other referenced fields like price and promotional price
+   */
+  protected updateCurrency(id: string, data: AdBasicData) {
+    this.currency = data.currencies.find(c => c.id === id || c.internalName === id);
+  }
+
+  /**
+   * Creates a list of categories with extra information (e.g level, leaf) used for rendering
+   */
+  protected populateCategories(items: AdCategoryWithChildren[], level: number) {
+    for (const item of items) {
+      const hasChildren = !empty(item.children);
+      const hierarchy: HierarchyItem = item as any;
+      hierarchy.level = level;
+      hierarchy.leaf = !hasChildren;
+      this.categories.push(hierarchy);
+      if (hasChildren) {
+        this.populateCategories(item.children, level + 1);
+      }
+    }
+  }
+
+  /**
+   * Resolves the current ad status label
+   */
+  get status(): string {
+    return this.marketplaceHelper.resolveStatusLabel((this.data as AdDataForEdit).status);
+  }
+
+  /**
+   * Creates or updates the current ad and allows to create
+   * a new ad based on the current one
+   */
+  save(insertNew?: boolean) {
+
+    validateBeforeSubmit(this.form);
+    if (!this.form.valid) {
+      return;
+    }
+
+    const value = this.form.value;
+
+    value.publicationPeriod = this.ApiHelper.dateRange(value.publicationBeginDate, value.publicationEndDate);
+    delete value['publicationBeginDate'];
+    delete value['publicationEndDate'];
+
+    value.promotionalPeriod = this.ApiHelper.dateRange(value.promotionalBeginDate, value.promotionalEndDate);
+    delete value['promotionalBeginDate'];
+    delete value['promotionalEndDate'];
+
+    const onFinish: any = (id: string) => {
+      this.notification.snackBar(this.i18n.ad.adSaved);
+      if (insertNew) {
+        EditAdComponent.BASED_ON_ID = id || this.id;
+        if (this.create) {
+          this.reload();
+        } else {
+          this.router.navigate(['/marketplace', this.owner, this.data.kind, 'new'], { replaceUrl: true });
```

Comment:
The router.navigate method support in the options the `queryParams` object. It should be used to pass the parameter to the new ad.